core/domain/exp_jobs_one_off.py

```diff
@@ -831,3 +832,100 @@ def reduce(key, values):
         # Combine all values from multiple lists into a single list
         # for that error type.
         yield (key, list(set().union(*final_values)))
+
+
+class ExplorationMigrationValidationJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job to validate exploration migration can be carried out
+    successfully.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted or item.states_schema_version != (
```

Comment:
You don't need the second check, right? When you load the exploration into a domain object, it automatically gets converted to the latest schema version I think (and you can assert that post-load).

At the very least, if you want to check this, you should yield rather than just ignore.